### PR TITLE
Shortcut improvements

### LINKED ISSUE #3583: Shortcut improvements
### Description of the Issue
In Shortcut mapper: 

-  we don't know which  category the shortcut belongs to
- we don't know which plugin the shortcut belongs to
- we don't see every shortcuts mapped in scintilla view.

### Expected Behavior

- add a 3rd column in the main menu that contains a category ("File", "Edit", "Format", "Search" ....)
- add a 3rd column in the plugin commands that contains the plugin name
- in the scintilla panel, when they are multiple mapped shortcut, print all of them with "or" separator.

Checkout the screenshots 
before pull request 
nppbeforeshortcut

after pull request
nppaftershortcut

### PR DESCRIPTION
**Pull request by cmeriaux**
Fixes #3583

Shorcut mapper - main panel : new colums that show the category of the shortcut
Shorcut mapper - plugin panel : new colums that show the plugin name that the shortcut belongs to
Shorcut mapper - scintilla panel : it shows every shortcuts configured for one command

### CODE REVIEW

File: PowerEditor/src/WinControls/shortcut/shortcut.h [hunk 1]
```diff
@@ -193,12 +194,42 @@ protected :
 		 
 class CommandShortcut : public Shortcut {
 public:
-	CommandShortcut(Shortcut sc, long id) :	Shortcut(sc), _id(id) {};
+	CommandShortcut(Shortcut sc, long id) :	Shortcut(sc), _id(id) {
+		long category_id = _id - _id % 500;
```
Comment (donho) [inline PowerEditor/src/WinControls/shortcut/shortcut.h]: why 500? Magic number should be avoided.

File: PowerEditor/src/WinControls/shortcut/shortcut.h [hunk 2]
```diff
@@ -193,12 +194,42 @@ protected :
...
+				_category = TEXT("Setting"); break;
+			case IDM_TOOL : 
+				_category = TEXT("Tool"); break; 
+			case IDM_EXECUTE : 
+				_category = TEXT("Execute"); break; 
+			default :
+				_category = TEXT(""); break;
+		}
+	};
 	unsigned long getID() const {return _id;};
```
Comment (donho) [inline PowerEditor/src/WinControls/shortcut/shortcut.h]: Function long definition should be in cpp file.

**Comment by donho** [comment]
@cmeriaux  Nice PR.
No need to redo the PR, please just add new commit for my change request in this PR.

**Comment by cmeriaux** [comment]
@donHo I've done your change request.

I noticed that some constants are not declared as expected.

Wrong code:
```cpp
    #define    IDM_EDIT_AUTOCOMPLETE                (50000 + 0)
    #define    IDM_EDIT_AUTOCOMPLETE_CURRENTFILE    (50000 + 1)
    #define    IDM_EDIT_FUNCCALLTIP                 (50000 + 2)
    #define    IDM_EDIT_AUTOCOMPLETE_PATH           (50000 + 6)

    #define    IDM_VIEW_GOTO_ANOTHER_VIEW        10001
    #define    IDM_VIEW_CLONE_TO_ANOTHER_VIEW    10002
    #define    IDM_VIEW_GOTO_NEW_INSTANCE        10003
    #define    IDM_VIEW_LOAD_IN_NEW_INSTANCE     10004
```
expected code:
```cpp
    #define    IDM_EDIT_AUTOCOMPLETE                (IDM_EDIT + 100)
    #define    IDM_EDIT_AUTOCOMPLETE_CURRENTFILE    (IDM_EDIT + 101)
    #define    IDM_EDIT_FUNCCALLTIP                 (IDM_EDIT + 102)
    #define    IDM_EDIT_AUTOCOMPLETE_PATH           (IDM_EDIT + 103)

    #define    IDM_VIEW_GOTO_ANOTHER_VIEW        (IDM_VIEW + 200)
    #define    IDM_VIEW_CLONE_TO_ANOTHER_VIEW    (IDM_VIEW + 201)
    #define    IDM_VIEW_GOTO_NEW_INSTANCE        (IDM_VIEW + 202)
    #define    IDM_VIEW_LOAD_IN_NEW_INSTANCE     (IDM_VIEW + 203)
```

I didn't change anything to avoid regression. Do I fix it ?

File: PowerEditor/src/WinControls/shortcut/shortcut.cpp [hunk 3]
```diff
@@ -955,3 +955,32 @@ INT_PTR CALLBACK ScintillaKeyMap::run_dlgProc(UINT Message, WPARAM wParam, LPARA
...
+		_category = TEXT("Format");
+	else if ( _id < IDM_ABOUT)
+		_category = TEXT("Lang");
+	else if ( _id < IDM_SETTING)
+		_category = TEXT("About");
+	else if ( _id < IDM_TOOL)
+		_category = TEXT("Setting");
+	else if ( _id < IDM_EXECUTE)
+		_category = TEXT("Tool");
+	else if ( _id < IDM_EXECUTE)
```
Comment (MAPJe71) [inline PowerEditor/src/WinControls/shortcut/shortcut.cpp]: Checking for `IDM_EXECUTE` twice?

**Comment by donho** [comment]
@cmeriaux  
> I didn't change anything to avoid regression. Do I fix it ?

No, please leave it as it is. They are using for the translation.

**Comment by donho** [comment]
@cmeriaux  please remove duplicated code.

**Comment by cmeriaux** [comment]
@donho @MAPJe71 code review done. Sorry for the typo.

**Comment by donho** [comment]
@cmeriaux  It seems you add your new commits on the old branch which has been merged since long time. It's bad because there'll be the merge problem on my side.
Please create a new branch for each PR, to avoid the merge problem.

**Comment by cmeriaux** [comment]
@donho  done in #3635

**Pull request closed without merging**